tests/queries/0_stateless/02473_functions_in_readonly_mode.sh

```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
+
+unset CLICKHOUSE_LOG_COMMENT
+
+CURDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+# shellcheck source=../shell_config.sh
+. "$CURDIR"/../shell_config.sh
+
+$CLICKHOUSE_CLIENT --query="SELECT * from numbers(1)"
+$CLICKHOUSE_CLIENT --query="SELECT * from format('TSV', '123')"
+
+$CLICKHOUSE_CLIENT --readonly=1 --query="SELECT * from numbers(1)"
+$CLICKHOUSE_CLIENT --readonly=1 --query="SELECT * from format('TSV', '123')" 2>&1 | grep -Fq "Cannot execute query in readonly mode. (READONLY)" && echo 'ERROR' || echo 'OK'
```

Comment:
@SmitaRKulkarni Why can't we execute this query in readonly mode?

Reply:
Only certain functions are allowed in readonly mode (`format` is not one of them) : https://github.com/ClickHouse/ClickHouse/pull/42708
This is an existing test. I used the same test file, changed it's extension to `.sh` and appended new insert tests

Reply:
Okay